model/user/user_test.go

```diff
@@ -754,6 +754,47 @@ func TestGeneralSubscriptionIDs(t *testing.T) {
 	}, u.GeneralSubscriptionIDs())
 }
 
+func TestViewableProject(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	env := testutil.NewEnvironment(ctx, t)
+	rm := env.RoleManager()
+
+	assert.NoError(t, db.ClearCollections(evergreen.RoleCollection, evergreen.ScopeCollection, Collection))
+	unrestrictedProjectScope := gimlet.Scope{
+		ID:        evergreen.UnrestrictedProjectsScope,
+		Name:      "unrestricted projects",
+		Type:      evergreen.ProjectResourceType,
+		Resources: []string{"mci", "spruce"},
+	}
+	err := rm.AddScope(unrestrictedProjectScope)
+	assert.NoError(t, err)
+
+	basicProjectAccessRole := gimlet.Role{
+		ID:          evergreen.BasicProjectAccessRole,
+		Name:        "basic access",
+		Scope:       evergreen.UnrestrictedProjectsScope,
+		Permissions: map[string]int{"project_tasks": 20, "project_patches": 10, "project_logs": 10, "project_task_annotations": 10},
```

Comment:
nit: can use the global values rather than hardcoding (such as evergreen.PermissionTasks)